types/index.d.ts

```diff
@@ -221,7 +221,10 @@ export interface FrozenProcessor<P = Settings> {
  * @typeParam P Processor settings
  * @returns Optional Transformer.
  */
-export type Plugin<S extends any[] = [Settings?], P = Settings> = Attacher<S, P>
+export type Plugin<
+  S extends any[] = [(Settings | boolean)?],
```

Comment:
@ChristianMurphy Hmm, this now results in TS thinking that a plugin can receive a boolean instead of `Settings?`.
I tried removing this change here, to no avail.

Do you know how I `(Settings|boolean)?` can be *passed*, but only `Settings?` received? 🤔

Reply:
Ping @ChristianMurphy

Reply:
For a single plugin, we'd want to/need to remove boolean from here and add it to

https://github.com/unifiedjs/unified/blob/5167c49ce7e02fcd518adae0acda80ccd4e97400/types/index.d.ts#L20-L23

would probably something like

```ts
 use<S extends any[] = [Settings?]>( 
   plugin: Plugin<S, P>, 
   ...settings: S | [boolean?]
 ): Processor<P> 
```